src/onelogin/saml2/idp_metadata_parser.py

```diff
@@ -60,16 +64,26 @@ def parse_remote(url):
         return OneLogin_Saml2_IdPMetadataParser.parse(idp_metadata)
 
     @staticmethod
-    def parse(idp_metadata):
+    def parse(idp_metadata, preferred_sso_binding=OneLogin_Saml2_Constants.BINDING_HTTP_REDIRECT):
         """
-        Parse the Identity Provider metadata and returns a dict with extracted data
-        If there are multiple IDPSSODescriptor it will only parse the first
+        Parse the Identity Provider metadata and return a dict with extracted data.
+
+        If there are multiple IDPSSODescriptor, parse only the first.
+
+        Parse only those SSO endpoints with the same binding as given by
+        the `preferred_sso_binding` parameter.
+
+        If there are multiple SSO endpoints, extract only the first.
```

Comment:
> Your current code is not exracting the first if there are multiple SSO endponts.

I left the corresponding logic untouched:

```
                    if len(sso_nodes) > 0:
                        idp_sso_url = sso_nodes[0].get('Location', None)
```

So it should, as before, extract the first one (with the specified preferred binding).